src/test/neorados/pool.cc

```diff
@@ -103,7 +104,8 @@ class NeoRadosPool : public CoroTest {
 
   /// \brief Delete pool used for testing
   boost::asio::awaitable<void> CoTearDown() override {
-    for (const auto& name : created_pools) try {
+    auto pools_to_delete = created_pools;
```

Comment:
Why can't we iterate over `created_pools`?

Reply:
It is required because created_pools is only modified by delete_pool not by delete_pool is calling via CoTearDown. I observed without this change a test case was getting crashed